graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/execution/FunctionDataFetcher.kt

```diff
@@ -86,7 +86,7 @@ open class FunctionDataFetcher(
      *
      *   - The entire environment is returned if the parameter is of type [DataFetchingEnvironment]
      */
-    protected open fun mapParameterToValue(param: KParameter, environment: DataFetchingEnvironment): Any? =
+    private fun mapParameterToValue(param: KParameter, environment: DataFetchingEnvironment): Any? =
```

Comment:
by keeping it open one could technically implement some other autowired object besides `GraphQLContext` and `DataFetchingEnvironment`

Reply:
That is interesting. Do wee need to leave this method in particular open or do we just have them re-implement `get` and call their own custom method.

Reply:
Since other methods are already open I'd keep those open as well. Once (or IF) we implement Kotlin native way of executing the queries (and drop the dependency on `CompletableFuture`) we could change the interface method signatures to something more appropriate.